src/Microsoft.AspNetCore.Server.IISIntegration/NativeMethods.cs

```diff
@@ -37,85 +29,212 @@ public enum REQUEST_NOTIFICATION_STATUS
+        {
+            register_callbacks(request_callback, shutdown_callback, managed_context_handler, pvRequestContext, pvShutdownContext);
+        }
+
+        public static unsafe int HttpWriteResponseBytes(IntPtr pInProcessHandler, HttpApiTypes.HTTP_DATA_CHUNK* pDataChunks, int nChunks, out bool fCompletionExpected)
+        {
+            return http_write_response_bytes(pInProcessHandler, pDataChunks, nChunks, out fCompletionExpected);
+        }
+
+        public static int HttpFlushResponseBytes(IntPtr pInProcessHandler, out bool fCompletionExpected)
+        {
+            return http_flush_response_bytes(pInProcessHandler, out fCompletionExpected);
+        }
+        public static unsafe HttpApiTypes.HTTP_REQUEST_V2* HttpGetRawRequest(IntPtr pInProcessHandler)
+        {
+            return http_get_raw_request(pInProcessHandler);
+        }
+
+        public static void HttpStopCallsIntoManaged()
+        {
+            http_stop_calls_into_managed();
+        }
+
+        public static void HttpStopIncomingRequests()
+        {
+            http_stop_incoming_requests();
+        }
+
+        public static unsafe HttpApiTypes.HTTP_RESPONSE_V2* HttpGetRawResponse(IntPtr pInProcessHandler)
+        {
+            return http_get_raw_response(pInProcessHandler);
+        }
+
+        public static void HttpSetResponseStatusCode(IntPtr pInProcessHandler, ushort statusCode, string pszReason)
+        {
+            Validate(http_set_response_status_code(pInProcessHandler, statusCode, pszReason));
+        }
+
+        public static unsafe int HttpReadRequestBytes(IntPtr pInProcessHandler, byte* pvBuffer, int cbBuffer, out int dwBytesReceived, out bool fCompletionExpected)
+        {
+            return http_read_request_bytes(pInProcessHandler, pvBuffer, cbBuffer, out dwBytesReceived, out fCompletionExpected);
+        }
+
+        public static void HttpGetCompletionInfo(IntPtr pCompletionInfo, out int cbBytes, out int hr)
+        {
+            http_get_completion_info(pCompletionInfo, out cbBytes, out hr);
+        }
+
+        public static void HttpSetManagedContext(IntPtr pInProcessHandler, IntPtr pvManagedContext)
+        {
+            Validate(http_set_managed_context(pInProcessHandler, pvManagedContext));
+        }
+
+        public static IISConfigurationData HttpGetApplicationProperties()
+        {
+            var iisConfigurationData = new IISConfigurationData();
+            Validate(http_get_application_properties(ref iisConfigurationData));
+            return iisConfigurationData;
+        }
+
+        public static bool HttpTryGetServerVariable(IntPtr pInProcessHandler, string variableName, out string value)
+        {
+            return http_get_server_variable(pInProcessHandler, variableName, out value) == 0;
+        }
+
+        public static bool HttpShutdown()
+        {
+            return http_shutdown();
```

Comment:
could not find this http_shutdown() native  method

Reply:
We can remove it; I wanted to be liberal with the native methods exposed as I didn't want to break the API.